docs/configuring-a-resource.md

```diff
@@ -22,126 +22,239 @@ cloud provider; sometimes it could simply be the name of resource
+model and Terraform configuration. We need a custom, per resource configuration
+to adapt Crossplane `external name` from Terraform `id`.
+
+![external name configuration](images/terrajet-externalname.png)
+
+Here are [the types for the External Name configuration]:
 
 ```go
+// SetIdentifierArgumentsFn sets the name of the resource in Terraform attributes map,
+// i.e. Main HCL file.
+type SetIdentifierArgumentsFn func(base map[string]interface{}, externalName string)
+// GetExternalNameFn returns the external name extracted from the TF State.
+type GetExternalNameFn func(tfstate map[string]interface{}) (string, error)
+// GetIDFn returns the ID to be used in TF State file, i.e. "id" field in
+// terraform.tfstate.
+type GetIDFn func(ctx context.Context, externalName string, parameters map[string]interface{}, providerConfig map[string]interface{}) (string, error)
+
 // ExternalName contains all information that is necessary for naming operations,
 // such as removal of those fields from spec schema and calling Configure function
 // to fill attributes with information given in external name.
 type ExternalName struct {
-    // SetIdentifierArgumentFn sets the name of the resource in Terraform argument
-    // map.
-    SetIdentifierArgumentFn SetIdentifierArgumentFn
-    
-    // OmittedFields are the ones you'd like to be removed from the schema since
-    // they are specified via external name. You can omit only the top level fields.
-    // No field is omitted by default.
-    OmittedFields []string
-    
-    // DisableNameInitializer allows you to specify whether the name initializer
-    // that sets external name to metadata.name if none specified should be disabled.
-    // It needs to be disabled for resources whose external name includes information
-    // more than the actual name of the resource, like subscription ID or region
-    // etc. which is unlikely to be included in metadata.name
-    DisableNameInitializer bool
+  // SetIdentifierArgumentFn sets the name of the resource in Terraform argument
+  // map. In many cases, there is a field called "name" in the HCL schema, however,
+  // there are cases like RDS DB Cluster where the name field in HCL is called
+  // "cluster_identifier". This function is the place that you can take external
+  // name and assign it to that specific key for that resource type.
+  SetIdentifierArgumentFn SetIdentifierArgumentsFn
+  
+  // GetExternalNameFn returns the external name extracted from TF State. In most cases,
+  // "id" field contains all the information you need. You'll need to extract
+  // the format that is decided for external name annotation to use.
+  // For example the following is an Azure resource ID:
+  // /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/mygroup1
+  // The function should return "mygroup1" so that it can be used to set external
+  // name if it was not set already.
```

Comment:
Perhaps you could say why this is needed. Why can't the external name be `/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/mygroup1`?

Reply:
While I appreciate the curiosity behind this question, I don't think it is something that worths mentioning in this guide. It answers the question "why did we do this?" rather than "how can I configure my resource?" and better fits a design document rather than a guide. 

Stating  this, please check [this comment](https://github.com/crossplane/terrajet/pull/132#issuecomment-966663178) to learn the reasoning behind.